dracut/30ignition/ignition-generator

```diff
@@ -92,13 +92,19 @@ if [ -n "$RANDOMIZE_DISK_GUID" ]; then
     add_requires "disk-uuid@${escaped_guid}.service"
 fi
 
-if [[ $(cmdline_arg coreos.oem.id) == "digitalocean" ]]; then
-    add_requires coreos-digitalocean-network.service
+oemid=$(cmdline_arg coreos.oem.id ${oem_id})
+# Compatibility Ignition 2 (impl spec 3) as created by coreos-assembler
+if [ -z "${oemid}" ]; then
+    platformid=$(cmdline_arg ignition.platform.id)
+    case "${platformid}" in
+        aws) oemid=ec2;;
+        gcp) oemid=gce;;
+        *) oemid=${platformid}
+    esac
 fi
 
-oem_id=pxe
-if [[ $(systemd-detect-virt || true) =~ ^(kvm|qemu)$ ]]; then
-    oem_id=qemu
+if [[ "${oemid}" == "digitalocean" ]]; then
+    add_requires coreos-digitalocean-network.service
 fi
 
-echo "OEM_ID=$(cmdline_arg coreos.oem.id ${oem_id})" > /run/ignition.env
+echo "OEM_ID=${oemid})" > /run/ignition.env
```

Comment:
Doesn't this have different functionality than before in that it will no longer default to `pxe`/`qemu`?

Reply:
Missed the commit message speaking about it. Resolving